astropy/utils/masked/tests/test_containers.py

```diff
@@ -81,6 +82,33 @@ def test_transformation(self):
         assert_array_equal(mgcrs.data.lon.unmasked, gcrs.data.lon)
         assert_array_equal(mgcrs.data.lat.unmasked, gcrs.data.lat)
 
+    def test_apply_space_motion(self):
+        # Regression test for gh-13041. It is important that the
+        # distance is missing here, so that a warning is raised.
+        kwargs = dict(
+            ra=[1, 2] * u.deg,
+            dec=[3, 4] * u.deg,
+            pm_ra_cosdec=[5, 6] * u.mas / u.yr,
+            pm_dec=[1, 2] * u.mas / u.yr,
+            obstime=Time(["2000-10-22T12:23:45", "2000-10-22T12:23:45"]),
+        )
+        sc = SkyCoord(**kwargs)
+        mask = np.array([False, True])
+        kwargs["pm_ra_cosdec"] = Masked(kwargs["pm_ra_cosdec"], mask=mask)
+        kwargs["pm_dec"] = Masked(kwargs["pm_dec"], mask=mask)
+        msc = SkyCoord(**kwargs)
+        new_time = Time("2020-10-22T12:23:45")
+        msg = 'yielded {} of "distance overridden'
+        with pytest.warns(erfa.ErfaWarning, match=msg.format(2)):
+            sc2 = sc.apply_space_motion(new_obstime=new_time)
```

Comment:
it feels a bit out of place to check for a warning that's emitted from another package. Shouldn't we *avoid* the warning, workaround it or silence it somehow ? I'm not sure what our actual options are, but I'm trying to put myself in the shoes of someone who's just calling `astropy` functions and classes, and gets a warning from something else. I think as an end user, this could look like a bug in astropy.